app/Http/Middleware/MultipleAppSupport.php

```diff
@@ -16,10 +17,15 @@ class MultipleAppSupport
     public function handle($request, Closure $next)
     {
         $requestDbName = strtolower($request->route('appName'));
-        $dbName = \Config::get('database.connections.mongodb.database');
+        $coreDbName = Config::get('sharedSettings.internalConfiguration.coreDatabaseName');
+        if ($requestDbName === $coreDbName) {
+            return $next($request);
+        }
+
+        $dbName = Config::get('database.connections.mongodb.database');
 
         if ($dbName !== $requestDbName) {
-            \Config::set('database.connections.mongodb.database', $requestDbName);
+            Config::set('database.connections.mongodb.database', $requestDbName);
```

Comment:
Missing check here if application exists, otherwise standard users can register / login to random application name